lua/rainbow/internal.lua

```diff
@@ -40,8 +47,19 @@ end
 
 local M = {}
 
+local function try_async(f)
+  if not f then return end
```

Comment:
I put in this line to stop the `f is nil` errors.
I think resolving why it's nil is related to why there's no updates after attach.

Reply:
i thought of that too. yay i'm thinking right!

Reply:
i'll make callbackfn(bufnr) return 0 or somehting?

Reply:
nah

Reply:
think I've fixed it